### ISSUE TITLE
Versions after 2.05.85 Sensor State Changes Lost to Home Assistant

### ISSUE THREAD

**Issue by majorsl** [dresden-elektronik/deconz-rest-plugin#3707]
## Describe the bug

I use deCONZ with the Home Assistant integration: https://www.home-assistant.io/integrations/deconz/

Upgrading to the beta 2.6.0 of deCONZ stops all my sensor states from changing in Home Assistant, but the states do still show in Phoscon.

I tried removing/re-adding the integration with no change. Of note, when I re-added the integration I had a device shown by HA that is "null" just () with no data or information.  This does not appear in Phoscon or deCONZ.  Erroneous data being sent by the API?

## Steps to reproduce the behavior

  If the problem is reproducable, list the steps here:

  1. Use Home Assistant
  2. Install a version after 2.05.85
  3. Watch your sensors no longer work in Home Assistant

## Expected behavior

Sensors toggle states in Home Assistant.

## Environment

 - Host system: Raspberry
 - Running method: Marthoc Docker container
 - Firmware version: (26660700)
 - deCONZ version: > greater than 2.05.85
 - Device: ConBee II
 - Do you use an USB extension cable: yes
 - Is there any other USB or serial devices connected to the host system? no

## Additional context

Has been working perfectly with Home Assistant up to this point. I have reverted away from the beta, but will be happy to try again if anything is found or a fix is available.

**Comment by Kane610** [comment]
It would be good if you could share debug logs from hass. Enable debug per deconz integration instructions and share them here

**Comment by Kane610** [comment]
Typically when no state changes are observed it's the websocket not setting up properly

**Comment by majorsl** [comment]
We have a thread going on reddit, someone did few tests and it seems the last version that is working is 2.05.84 (I think that is 85). I'm above to revert to that if I can figure out how to use a local image I still have - never done that. https://www.reddit.com/r/homeassistant/comments/jx7lxo/0118_and_deconz_phoscon_issues/

**Comment by majorsl** [comment]
Confirmed: 2.05.85 is the last version that worked. Stepping back through the Docker container versions one by one to it finally started sending sensor states to Home Assistant again.

**Comment by Mimiix** [comment]
I have no issues on 2.60.0 with HA and deCONZ. Running deCONZ in a seperate VM.

**Comment by manup** [comment]
I've looked in the GIT diff between v2.5.85 and v2.5.86, the only REST-API difference here seems to be:

![image](https://user-images.githubusercontent.com/383386/99782169-19164a00-2b19-11eb-98a6-e38518b1234a.png)

Which is related to ELKO Super TR:

```
{
  "config": {
       "temperaturemeasurement" : "air sensor"
  }
}
```

Could it be that?

**Comment by SwoopX** [comment]
@manup Already discussed with @Kane610 . That's why I updated the API documentation and added when something new was introduced. Cannot be sure however, that I caught everything.

I guess a bit more was introduced.

**Comment by manup** [comment]
According to the diff there seems to only the above and one error message which is different in the REST-API output.

![image](https://user-images.githubusercontent.com/383386/99784310-ea4da300-2b1b-11eb-9f77-62fd77cd0889.png)

Otherwise I can't see anything :thinking:

**Comment by SwoopX** [comment]
A bit more. Besides the config part already mentioned, also this
![grafik](https://user-images.githubusercontent.com/4005212/99785181-17e71c00-2b1d-11eb-8478-694e35dba0a0.png)

**Comment by manup** [comment]
So the question is if these devices are paired in the problematic installations?

**Comment by manup** [comment]
The PR seems to only add new attributes but doesn't change existing REST-API attributes. This hopefully shouldn't break a client.

**Comment by Kane610** [comment]
The deconz integration creates standard sensors if there isn't added explicit support for the sensor so it shouldn't break but it might not expose the best information about it

**Comment by indomus** [comment]
The same problem I have.

**Comment by indomus** [comment]
Same problem here. https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3645

**Comment by majorsl** [comment]
> Same problem here. #3645

My goal yesterday was to get up and running as quickly as possible.  What can I provide that hasn't been discussed in the above issue to try to help?

I'm using Docker so it isn't too much of an issue to create latest or stable to get some information and then I can switch back to a working environment.

Fortunately, I don't have many devices and no repeaters yet, so if it is a device causing it maybe changes there can be a clue. I have these: SNZB-04 (6), Xiaomi Aqara Temp/Humid Sensor (1), CentraLite 3323-G (3)

**Comment by adrha** [comment]
Hey @manup first of all thank you for your great support!

I'm having the same issues as others here in this thread but recently figured out that the .85-version is the first bugy-release (and not 86 as reported earlier).
For me a roll-back to 84 fixed it.
The interesting point is, that others in this thread are using Aqara Temp/Humidity sensors, which i also use and they are having issues as well.

**Comment by Florian-Schmidt** [comment]
Same issue here: Suddenly HA automations stopped working during the last days.

Tried to update deconz and ConBee II to:
deconz: 2.06.00
firmware: 26670700

Every device was not reachable, over the day some devices came back. Now downgraded back to.
deconz: 2.05.86
firmware: 26660700

Still the same, updated deconz again to 2.06.00 but no changes. Also tried an older version 2.05.73 but still the same devices available. 

Unfortunately I don't have a 2.05.84 available.

**Comment by Florian-Schmidt** [comment]
Ok, build marthoc/deconz container locally and now running.

deconz: 2.05.84
firmware: 26650700

But still the same, there are some connections shown in VNC viewer but many devices are missing and the zigbee network doesn't build up.

**Comment by Florian-Schmidt** [comment]
Ok, now the first lights are coming back again and zigbee network grows up slowly...

Anyone an idea how to speed up the reconnect of known devices?

**Comment by SwoopX** [comment]
@Florian-Schmidt Please do not hijack other issues. You mentioned that none of your devices are reachable, which is a completely different topic so for this, please raise your own issue.

To answer your question: You can press 0 (zero) in deconz GUI while selecting a routing node, or F5.

**Comment by Florian-Schmidt** [comment]
@SwoopX Thanks for you answer. Sorry, thought this could be helpful as these releases seem to have several similar issues...

Actually Zigee network seems to continue growing up more and more. Going to wait some more days and hope to reach complete network again. In case of getting disappointed by incomplete network, I'll open new ticket.

**Comment by majorsl** [comment]
> As there hasn't been any response in 21 days, this issue has been automatically marked as stale. **At OP: Please either close this issue or keep it active** It will be closed in 7 days if no further activity occurs.

Still an issue.

**Comment by adrha** [comment]
Has someone got it to work from you guys?
I'm still running on .84 version. The latest version does still not work with  HomeAssistant.
Is there a way to export the device-config and import it on the new setup without repairing all nodes?

**Comment by majorsl** [comment]
> Has someone got it to work from you guys?
> I'm still running on .84 version. The latest version does still not work with HomeAssistant.
> Is there a way to export the device-config and import it on the new setup without repairing all nodes?

Not that I'm aware. I was going to try it again today based on #3977 which seems similar.  I had not waited hours for the sensors to come back so i thought I'd give that a go.

I agree - rebuilding from scratch just over an upgrade isn't optimal, but the later builds have some device support I want to use. :(

**Comment by thecem** [comment]
Which HA Version 2020.12.1?
Tried Phoscon Version 2.07.01?
Tried latest Conbee FW II 26680700?

**Comment by majorsl** [comment]
> Which HA Version 2020.12.1?
> Tried Phoscon Version 2.07.01?
> Tried latest Conbee FW II 26680700?

Yes. Already was running HA 2020.12.1.  I just updated to the latest stable version and upgraded my firmware to 26680700.

Again, the same issue where the sensors trigger in deCONZ and Phoscon, but _do not_ register in Home Assistant.  Lights work okay.

As mentioned in #3977, I am going to wait 6-8 hours and see if they eventually start to work again.  If not, I will revert to the last known working deCONZ.

**Comment by thecem** [comment]
did you try the services in HA:

`deconz.device_refresh`

or 

`deconz.remove_orphaned_entries`

**Comment by majorsl** [comment]
> did you try the services in HA:
> 
> `deconz.device_refresh`
> 
> or
> 
> `deconz.remove_orphaned_entries`

I just tried both, still no status updates in Home Assistant.

**Comment by Mimiix** [comment]
@Kane610 can you pitch in and see if this is something on our end?

**Comment by Kane610** [comment]
Is there changes to how the docker is configured because this typically means that the websocket isn't set up correctly

**Comment by majorsl** [comment]
> Is there changes to how the docker is configured because this typically means that the websocket isn't set up correctly

On my end, no.  I am using Portainer to work with deCONZ.  I shut down 2.05.85, did a Duplicate/Edit of the container and changed it to marthoc/deconz:stable and started it.  No changes whatsoever other than the latest stable version.

I believe the web socket is working because I can control (for example) an outlet with Home Assistant just fine. It is just _sensors_ that are not working which makes my issue identical to #3977.

**Comment by Kane610** [comment]
Sending command and receiving updates are two different things. Does state updates get reflected to outlets and lights?

**Comment by majorsl** [comment]
> Enable debug logs in hass for deconz integration.

I don't know how to do this and I don't see an option in the GUI for that.

**Comment by majorsl** [comment]
Also, I don't know if it means the same thing, if I toggle a light on/off in Phoscon I do see the state change in Home Assistant GUI.  So HA is receiving what I call an "update" to those devices.

**Comment by Kane610** [comment]
> > Enable debug logs in hass for deconz integration.
> 
> I don't know how to do this and I don't see an option in the GUI for that.

Check deconz integration documentation for how to enable debug

**Comment by adrha** [comment]
In my setup i use some Aqara Opple-switches (8CH).
After updating to latest or stable, the events of button-presses are not forwarded to homeassistant anymore.
Also temperature/humidity-sensors-updates are not forwarded anymore.
When querying through GET-Request on the API the updated values of the sensors are returned.

So obviously just the websocket-part is faulty.

As far as i know, there is an option to manually download the deconz-config in the frontend.
After uploading this to a fresh installation, do i have go re-pair all devices or should it work directly?

**Comment by majorsl** [comment]
> In my setup i use some Aqara Opple-switches (8CH).
> After updating to latest or stable, the events of button-presses are not forwarded to homeassistant anymore.
> Also temperature/humidity-sensors-updates are not forwarded anymore.
> When querying through GET-Request on the API the updated values of the sensors are returned.
> 
> So obviously just the websocket-part is faulty.
> 
> As far as i know, there is an option to manually download the deconz-config in the frontend.
> After uploading this to a fresh installation, do i have go re-pair all devices or should it work directly?

If you try it and it works, let me know.  I too have a temp/humidity sensor and it is also not working with Home Assistant. Something clearly has changed/broken in these later versions with many people reporting the same thing in several issues here and via replies to them.

**Comment by majorsl** [comment]
I enabled the debug log in HA.  This is what I see when I use Phoscon to toggle the plug near the couch "on".

2020-12-28 12:42:44 DEBUG (MainThread) [pydeconz.websocket] {"e":"changed","id":"2","r":"lights","state":{"alert":null,"on":true,"reachable":true},"t":"event","uniqueid":"00:12:4b:00:1e:72:03:06-01"}
2020-12-28 12:42:44 DEBUG (MainThread) [pydeconz.websocket] {"e":"changed","id":"65520","r":"groups","state":{"all_on":true,"any_on":true},"t":"event"}
2020-12-28 12:42:44 DEBUG (MainThread) [pydeconz.websocket] {"e":"changed","id":"1","r":"groups","state":{"all_on":true,"any_on":true},"t":"event"}
2020-12-28 12:42:45 DEBUG (MainThread) [pydeconz.websocket] {"attr":{"id":"2","lastannounced":null,"lastseen":"2020-12-28T17:42Z","manufacturername":"SONOFF","modelid":"S31 Lite zb","name":"Couch","swversion":null,"type":"On/Off plug-in unit","uniqueid":"00:12:4b:00:1e:72:03:06-01"},"e":"changed","id":"2","r":"lights","t":"event","uniqueid":"00:12:4b:00:1e:72:03:06-01"}

I then tripped two door/window sensors.  Their state changed in Phoscon, but _not in Home Assistant_.  **There were no logs generated for the sensor events.**

**Comment by Kane610** [comment]
@Mimiix some events reach hass but not others so it sounds like something on deconz side

**Comment by majorsl** [comment]
@Kane610 thank you for helping me get in the right direction for the logs.

**Comment by Mimiix** [comment]
@majorsl can you provide some logs?

To enable logging. Open deCONZ and click Help. After that, click Debug View.  The following debug levels need to be enabled for proper logging: INFO, ERROR,ERROR_L2,APS,APS_L2.

**Comment by s0ftice** [comment]
@Mimiix I provided logs a week ago here https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3977

**Comment by Mimiix** [comment]
Please don't hijack others issues...

**Comment by majorsl** [comment]
@Mimiix we do have same issue, perhaps his will help. After enabling, do you want me to do the same tests I did for Mimiix?  Toggle a light/plug then the sensors?  Thanks for helping.

**Comment by s0ftice** [comment]
We believe it's the same

**Comment by Mimiix** [comment]
> @Mimiix we do have same issue, perhaps his will help. After enabling, do you want me to do the same tests I did for Mimiix?  Toggle a light/plug then the sensors?  Thanks for helping.

Yes.

Your symptoms are the same, but we can't be sure yet. What else do your setups share   other than the same problem?

For example: I have ikea sensors and xiaomi sensors working and I have no issues at all. If it is that widespread, more should be affected.

**Comment by majorsl** [comment]
@Mimiix I'm running this in Docker.  I can get into debug and see the logging, but I can't save it or copy/paste out of VNC to get them here.  I can probably screenshot the sections and past in numbered order here, unless you have a better option...

**Comment by majorsl** [comment]
@Mimiix & @Kane610 ...and now everything is working all at once again, just like @s0ftice said in his issue.  I did nothing.

About 5 hours since I started up deCONZ.  Interesting that in my Time Zone I'm UTC-05:00.

**Comment by Mimiix** [comment]
> @Mimiix I'm running this in Docker.  I can get into debug and see the logging, but I can't save it or copy/paste out of VNC to get them here.  I can probably screenshot the sections and past in numbered order here, unless you have a better option...

Use vnc to connect to it?

**Comment by s0ftice** [comment]
I propose to mark mine (https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3977) as duplicate of this issue. As @majorsl observed the same magic "self-healing" after a few hours.

@Mimiix agree, this does not seem to be widespread and unfortunately, I don't see what our setups have in common which would easily explain what causes the issue. My setup is as vanilla as possible (see environment of 3977).

**Comment by SwoopX** [comment]
Hm, interesting observation with the time zone. Anybody willing to change it (e.g. from UTC-5 to UTC-1) and double check, if functionality resumes after 1 hour?

**Comment by majorsl** [comment]
I will be able to try this in a few days. Right now, everyone is home for the holiday and taking down home-automation for testing isn't a good idea. ;)

**Comment by s0ftice** [comment]
I'm at UTC+8 and it still took 5-6 hours. I can try to conduct the following tests.
Figure out if time to recover is
1. constant
2. dependent on the current time when I restart deCONZ

**Comment by Kane610** [comment]
Try upgrading to latest hass release there is a minor improvement to Deconz integration websocket. Would be interesting if it would affect this

**Comment by adrha** [comment]
Okay yesterday i updated deconz to _stable_ and the described issues came back.
After leaving it like this for a few hours, all devices came back (also recognized by someone in this thread).
I now updated hass to the stable-release and everything works after starting hass (without waiting for devices).
I don't know if the time after the update fixed it or the actual release from today.
I will keep an eye on it while updating in the coming days and weeks :)

**Comment by Kane610** [comment]
Do you see any errors in hass logs?

**Comment by adrha** [comment]
> Do you see any errors in hass logs?

Nope there are no errors or other logs which are related to deconz :)

**Comment by s0ftice** [comment]
I just updated Home Assistant Core from 2020.12.1 to 2020.12.2 and restarted deCONZ. It did not fix the issue unfortunately. @adrha unfortunately, I assume it was the time that fixed it.
Easy to find out: restart deCONZ :)

**Comment by s0ftice** [comment]
I think I found the root cause. I still don't know why it happens, but I got an easy workaround!

I measured the time it took to "magically heal" the situation, it was exact 8 hours. And I do live in Singapore (UTC+08), so can't be a coincidence.
I simply did the following:

`sudo dpkg-reconfigure tzdata`
Changed it to Europe -> London (which is GMT or UTC+00), then
`sudo systemctl restart deconz-gui`

and guess what: HA gets all event right after the deCONZ restart!
(my set-up is HA on a VM and deCONZ on a RPi3 using the https://phoscon.de/en/conbee/sdcard image)

The workaround is not ideal, but much better than waiting for 8 hours ;)

@Mimiix and team, please help to mark https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3977 as duplicate.

Thanks

**Comment by Mimiix** [comment]
Okay but now what?

**Comment by s0ftice** [comment]
well, please investigate.
good starting point, use your own image and set the time zone to Singapore (SGT) during installation and try to reproduce.
furthermore, analyse the code for checks related to time zone...

**Comment by SwoopX** [comment]
> 
> 
> I think I found the root cause. I still don't know why it happens, but I got an easy workaround!
> 
> I measured the time it took to "magically heal" the situation, it was exact 8 hours. And I do live in Singapore (UTC+08), so can't be a coincidence.
> I simply did the following:
> 
> `sudo dpkg-reconfigure tzdata`
> Changed it to Europe -> London (which is GMT or UTC+00), then
> `sudo systemctl restart deconz-gui`
> 
> and guess what: HA gets all the (sensor and wireless switches) events right after the deCONZ restart!
> (my set-up is HA on a VM and deCONZ on a RPi3 using the https://phoscon.de/en/conbee/sdcard image)
> 
> The workaround is not ideal, but much better than waiting for 8 hours ;) Phew!
> 
> @Mimiix and team, please help to mark #3977 as duplicate.
> 
> Thanks

Thanks for having done the round trip, that's what I was basically after. So we do have a timezone/timestamp topic here. Happy to quickly test it in my dev environment.

**Comment by Mimiix** [comment]
> well, please investigate.
> good starting point, use your own image and set the time zone to Singapore (SGT) during installation and try to reproduce.
> furthermore, analyse the code for checks related to time zone...
> 
> 


My tag to those involved was missing.

I am wondering if this is something that needs to be fixed on the side of ha or deconz.

@SwoopX @Kane610?

If it's on deconz side, I'll ask Manuel.

**Comment by SwoopX** [comment]
Stuff is happily continuing to drip in

![grafik](https://user-images.githubusercontent.com/4005212/103412306-b7fc8000-4b74-11eb-91d5-56938397fbfd.png)

![grafik](https://user-images.githubusercontent.com/4005212/103412318-d6fb1200-4b74-11eb-9995-da0d8b1841de.png)

**Comment by Kane610** [comment]
> > well, please investigate.
> > good starting point, use your own image and set the time zone to Singapore (SGT) during installation and try to reproduce.
> > furthermore, analyse the code for checks related to time zone...
> 
> My tag to those involved was missing.
> 
> I am wondering if this is something that needs to be fixed on the side of ha or deconz.
> 
> @SwoopX @Kane610?
> 
> If it's on deconz side, I'll ask Manuel.

As stated before this is on deconz side. If it is with getting devices available and time stamp issue it can't be hass

**Comment by Mimiix** [comment]
> > > well, please investigate.
> > > good starting point, use your own image and set the time zone to Singapore (SGT) during installation and try to reproduce.
> > > furthermore, analyse the code for checks related to time zone...
> > 
> > 
> > My tag to those involved was missing.
> > I am wondering if this is something that needs to be fixed on the side of ha or deconz.
> > @SwoopX @Kane610?
> > If it's on deconz side, I'll ask Manuel.
> 
> As stated before this is on deconz side. If it is with getting devices available and time stamp issue it can't be hass

Wondering if Homebridge hue is affected then. @ebaauw, can you answer that?

**Comment by s0ftice** [comment]
@SwoopX thanks for checking. It's probably a combination of two or more "circumstances", of which the time zone seems to be one. If it's just the time zone, there would have been many more reports...

I'm curious if the same workaround will fix it for @adrha, @majorsl, @indomus - but pretty sure it will.
@Florian-Schmidt, I think yours is different, as you described it suddenly stopped working (or was that related to a restart of deCONZ too?)

**Comment by SwoopX** [comment]
@s0ftice No problem. Happy to check whatever I can.

@Kane610 Honestly, what can I do more? Tell me what to do and I do it. I reconfigured my system and the screenshots evidence all is working with the small difference that I do not use HA. And this seems to be toe only deviating detail here. Again, I'm still open it could be anything on the deconz end... However, from my perspective, air is getting a bit thin for the argument "this is on deconz side. If it is with getting devices available and time stamp issue it can't be hass".

Admit, haven't changed the time zone in deconz yet. Will check it now.

**Comment by adrha** [comment]
That's funny with the time-zone.
I remember, everything worked after an hour or two. I live in Switzerland and we actually have UTC+1h :)
I've checked the time in both containers (i'm running it dockered) and they both have the localtime as time.

**Comment by Kane610** [comment]
> @s0ftice No problem. Happy to check whatever I can.
> 
> @Kane610 Honestly, what can I do more? Tell me what to do and I do it. I reconfigured my system and the screenshots evidence all is working with the small difference that I do not use HA. And this seems to be toe only deviating detail here. Again, I'm still open it could be anything on the deconz end... However, from my perspective, air is getting a bit thin for the argument "this is on deconz side. If it is with getting devices available and time stamp issue it can't be hass".
> 
> Admit, haven't changed the time zone in deconz yet. Will check it now.

What I mean is that I don't care about timestamps or time zones or anything on hass side. All I get are events from Deconz, if I don't get events but websocket is setup it should be something on deconz side.

**Comment by majorsl** [comment]
Mine is working. Party due to a Time Zone issue and mostly due to myself.

It seems versions before the ones noted in this ticket didn't care (or possibly use) the Docker TZ env variable. In looking to change my Docker to use Europe/London to test, I noticed I had  America/New York... which is a typo on my part; should be America/New_York

As such, Docker was not using it nor throwing an obvious error.  Whenever I looked at the logs, deCONZ, Docker, etc all the times were shown to be correct so I just never thought about that one variable.

With that said, there does appear to be something with the Time Zone and sensors for those that may not have a Time Zone configured exactly as expected.

_Note to self: get in the habit of using America/Boston instead._

**Comment by ebaauw** [comment]
> Wondering if Homebridge hue is affected then. @ebaauw, can you answer that?

I haven't seen any issues similar to this with Homebridge Hue and deCONZ.  Of course, Homebridge Hue, like Phoscon uses polling next to the web socket  (because that's the only way to work with a Hue bridge).  However, looking at my logs, the sensor state changes are coming in through the web socket.

As I suggested in Discord, can you please try another web socket client in parallel to HASS, to see if the problem is with generating the events, or sending the events.  Homebridge Hue comes with a `dc_eventlog` utility, but anything similar would do.  Also, please check the deCONZ log: it should issue a message for each web socket event sent to each client.

**Comment by Florian-Schmidt** [comment]
> @Florian-Schmidt, I think yours is different, as you described it suddenly stopped working (or was that related to a restart of deCONZ too?)

@s0ftice It was related to an update. Actually running 2.08.00.

**Comment by cmuellner** [comment]
I was running into this today as well (client: home-assistant on an external machine; deconz environment: raspi3/Fedora/podman).
My client happily found all devices and I could enable/disable all lights controlled by deconz. However, sensor updates were detected and logged in deconz, but never forwarded to the client.
Setting the timezone to Europe/Londong ("timedatectl set-timezone Europe/London") and restarting the container immediately fixed the issue.

**Comment by Metus88** [comment]
same problem for me...

**Comment by JohCo-hassio** [comment]
Same problem for me.

**Comment by Mimiix** [comment]
For anyone willing to add a #metoo : Please do not. This clutters communication and my mailbox. 

It seems to be affected by timezone. I am currently checking if this is something we should fix or HA should fix. Iam checking with Manuel.

**Comment by manup** [comment]
By looking at the code I think the problem is that the push logic depends very strictly on correct time to be set — at all times.
Even when the time settings are corrected, the former timestamps of sensor states are carried on.

This is not unique to HA or any client. Having incorrect time settings can happen for various reasons, and the bigger problem is that even correcting them won't fix the problem immediately until all timestamps have been refreshed.

I think to protect the push logic from such issues it needs to be completely independent from any absolute timestamps.

**Comment by s0ftice** [comment]
Thanks @manup. I don't think that's the main issue to be honest. My HA VM as well as my RPi with deCONZ use ntp to sync their time, so they are set correctly and were also set to the correct time zone - now my RPi's time zone is set to GMT (wrongly) to make it work.

Having to wait a full 8h (exactly my UTC offset) after every deCONZ restart for the sensor events to be pushed out, indicates that it's 100% related to the time zone setting (for users who are affected - still no clue why not everyone is facing this problem). And after those 8h of wait time (exact to the second!), it starts working reliably and continuously (until the next deCONZ restart).

@Mimiix yes, the #metoo clutter the Inbox, but also gives a good indication of this being a slightly broader issue.

Thanks again for looking into this.

**Comment by Florian-Schmidt** [comment]
Strange yesterday evening suddenly the lights stopped working again. I thought that both containers (deconz and homesssistant) are running with correct timezone.

Now set environment variable to use TZ=Europe/Berlin for sure but still no chance to get the lights controllable again. I'll check in hours to see if there's an relation to the timezone offset.

**Comment by Florian-Schmidt** [comment]
@s0ftice Seems as you are right. Yesterday one hour later the lights were controllable via HA again but later they were not. Perhaps there are more reasons as only the timezone.

Docker logs show two messages.

During standby: 
0x00158D000418D399 error APSE-DATA.confirm: 0xE1 on task

When pushed HA button: 
delay sending request 213 dt 1 ms to 0X00158D000418D399, ep: 0x01 cluster:0x0006 on Air: 1

**Comment by Mimiix** [comment]
> @s0ftice Seems as you are right. Yesterday one hour later the lights were controllable via HA again but later they were not. Perhaps there are more reasons as only the timezone.
> 
> Docker logs show two messages.
> 
> During standby:
> 0x00158D000418D399 error APSE-DATA.confirm: 0xE1 on task
> 
> When pushed HA button:
> delay sending request 213 dt 1 ms to 0X00158D000418D399, ep: 0x01 cluster:0x0006 on Air: 1

The 0xE1 is a error message. It basically means you are suffering from interference issues.

**Comment by Florian-Schmidt** [comment]
@Mimiix What should this mean exactly? It stopped working with newer Deconz versions and downgrades don‘t help. Two days ago the lights worked and without changes stopped to react.

VNC shows connections between the Zigbee devices. 

I‘ve got no idea where this is actually coming from...

**Comment by Mimiix** [comment]
Are you using a extension cable? Interference comes from USB3. 0 or wifi.

**Comment by Florian-Schmidt** [comment]
> Are you using a extension cable? Interference comes from USB3. 0 or wifi.

Added an short extension cable much earlier as the problems started. This seemed to be an advantage in other threads. USB 3.0 ports left earlier because of experiences from other users and strange situations.

Honestly, appreciate every idea and possible solution but this should be the reasons to stop working suddenly... We are in 2021 now and these reasons shouldn't make our life harder, they should improve.

**Comment by majorsl** [comment]
> > Are you using a extension cable? Interference comes from USB3. 0 or wifi.
> 
> Added an short extension cable much earlier as the problems started. This seemed to be an advantage in other threads. USB 3.0 ports left earlier because of experiences from other users and strange situations.
> 
> Honestly, appreciate every idea and possible solution but this should be the reasons to stop working suddenly... We are in 2021 now and these reasons shouldn't make our life harder, they should improve.

I don't know if this will help, but it is worth asking. Do you see any new 802.11b/g/n networks around you? Did you change your own?  If you can, get a wifi scanner for your phone. I use WiFi analyzer on my Android phone.

I had a conflict with Wifi and I had to move my Zigbee over to channel 25.  I found this helpful to sort out where in the spectrum things were: https://www.metageek.com/training/resources/zigbee-wifi-coexistence.html

**Comment by Mimiix** [comment]
> > Are you using a extension cable? Interference comes from USB3. 0 or wifi.
> 
> Added an short extension cable much earlier as the problems started. This seemed to be an advantage in other threads. USB 3.0 ports left earlier because of experiences from other users and strange situations.
> 
> Honestly, appreciate every idea and possible solution but this should be the reasons to stop working suddenly... We are in 2021 now and these reasons shouldn't make our life harder, they should improve.

Blame Intel on the interference issues. Also: the ssd probsbly needs shielding.

**Comment by Florian-Schmidt** [comment]
> I don't know if this will help, but it is worth asking. Do you see any new 802.11b/g/n networks around you? Did you change your own? If you can, get a wifi scanner for your phone. I use WiFi analyzer on my Android phone.
> 
> I had a conflict with Wifi and I had to move my Zigbee over to channel 25. I found this helpful to sort out where in the spectrum things were: https://www.metageek.com/training/resources/zigbee-wifi-coexistence.html

Nothing changed in my own WiFi and networks around us are not that much. But perhaps WiFi issues are an idea to check and I'm going to check your link...

**Comment by s0ftice** [comment]
@Florian-Schmidt I think yours is different. The issue described here is not about unavailable lights directly, it's about sensors and battery-based switches (that might be triggering lights). Once they become available after the timezone delta to UTC+00 has been reached, they are working reliably. You said "later they were not": this is not the case for us facing this issue (unless we restart deCONZ).

**Comment by Florian-Schmidt** [comment]
@s0ftice I'm going to create a new ticket my issues are changing daily at the moment. Actually it seems not to use expected timezone and suddenly web interface is rejecting password.
